lightning/src/onion_message/messenger.rs

```diff
@@ -196,7 +196,7 @@ impl<Signer: Sign, K: Deref, L: Deref> OnionMessenger<Signer, K, L>
 				Hmac::from_engine(hmac).into_inner()
 			};
 			match self.keys_manager.ecdh(Recipient::Node, &msg.onion_routing_packet.public_key,
-				Some(&blinding_factor))
+				Some(&Scalar::from_be_bytes(blinding_factor).unwrap()))
```

Comment:
Should we be mapping the `err` here instead of `unwrap`ing? Perhaps it's guaranteed to succeed anyway.. Similar in `chan_utils` and `onion_utils`

Reply:
(because the chance of failure is `2 ** -128`, so more likely you have a bug or hardware failure)